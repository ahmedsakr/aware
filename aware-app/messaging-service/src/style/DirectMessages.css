#direct-messages {
    background-color: #a2abab;
    display: block;
}

#direct-messages > input {
    margin-bottom: 0.5em;
    height: 35px;
    font-size: 20px;
    width: 85%;
}

#right-bar > a {
    text-align: center;
}

#courses {
  text-align: left;
  padding-left: 8%; 
  padding-top: 0.5em;
  padding-bottom: 0.5em;
  color: white;
  font-size: 20px;
  margin: 0;
}
