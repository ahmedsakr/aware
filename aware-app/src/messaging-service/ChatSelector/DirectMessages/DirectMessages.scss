/*
 * Direct Messages styling
 *
 * Defines layout fundamentals for Direct Messages.
 */

#direct-messages {
    display: flex;
    flex-direction: column;

    >input {
        margin-bottom: 0.5em;
        height: 35px;
        align-self: center;
        flex: 0;
    }
}

#direct-messages-new {
    background: rgb(233, 152, 3);
    color: white;
    display: flex;
    justify-content: center;
    align-items: center;
    margin-bottom: 10px;
    opacity: 0.9;

    * {
        padding: 5px;
    }

    > p {
        margin: 0;
    }

    &:hover {
        opacity: 1;
    }
}

#right-bar>a {
    text-align: center;
}