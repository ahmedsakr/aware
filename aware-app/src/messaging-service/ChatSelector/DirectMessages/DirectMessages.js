import React, { Component } from 'react';
import './DirectMessages.scss';
import DirectMessage from '../DirectMessage/DirectMessage'

class DirectMessages extends Component {
  render() {
    return (
      <div id="direct-messages" class="row">
        <p id="courses" class="col-12">Private Messages</p>
        <input id="textfield" placeholder="Search Messages..." />
        <DirectMessage src="/icons8-user-80.png" name="Josh Campitelli" />
        <DirectMessage src="/icons8-user-80.png" name="Arsalan Sadiq" />
        <DirectMessage src="/icons8-user-80.png" name="Ahmed Sakr" />
        <DirectMessage src="/icons8-user-80.png" name="TheRealBoys" />
      </div>
    );
  }
}

export default DirectMessages;
