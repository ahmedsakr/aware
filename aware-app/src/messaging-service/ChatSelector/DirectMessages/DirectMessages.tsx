import React from 'react';
import './DirectMessages.scss';
import DirectMessage from '../DirectMessage/DirectMessage'
import {Room} from '../ChatSelector'

type DirectMessagesProps = {
    selectDirectMessage: (directMessage: Room) => void
};

type DirectMessagesState = {};

export default class DirectMessages extends React.Component<DirectMessagesProps, DirectMessagesState> {
    render() {
        return (
            <div id="direct-messages">
                <input id="textfield" placeholder="Search Messages..." />
                <DirectMessage
                    selectDirectMessage={this.props.selectDirectMessage}
                    room=""
                    src="/icons8-user-80.png"
                    name="Josh Campitelli" />
            </div>
        );
    }
}
