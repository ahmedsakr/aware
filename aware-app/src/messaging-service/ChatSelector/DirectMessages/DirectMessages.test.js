import React from 'react';
import { shallow } from 'enzyme';

import DirectMessages from './DirectMessages';

describe('DirectMessages', () => {
  it('should render correctly in "debug" mode', () => {

    const component = shallow(<DirectMessages debug />);
  
    expect(component).toMatchSnapshot();
  });
});