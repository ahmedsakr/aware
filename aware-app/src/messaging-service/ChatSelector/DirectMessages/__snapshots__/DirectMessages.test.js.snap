version https://git-lfs.github.com/spec/v1
oid sha256:f4217c2c91eae6643a673e32a827da454f27e410c80098dc5fe41d8cfdd161ef
size 618
